e2e/test/scenarios/visualizations/table.cy.spec.js

```diff
@@ -269,6 +270,46 @@ describe("scenarios > visualizations > table", () => {
     cy.wait(100);
     popover().should("not.exist");
   });
+
+  it("popover should not be horizontally scrollable (metabase#31339)", () => {
+    openPeopleTable();
+    headerCells().filter(":contains('Password')").click();
+
+    popover().within(() => {
+      cy.findByText("Filter by this column").click();
+      cy.findByPlaceholderText("Search by Password").type("e").blur();
+      cy.wait("@findSuggestions");
+    });
+
+    popover().then(popoverElement => {
+      expect(
+        popoverElement[0].clientHeight,
+        "horizontal scrollbar is not shown",
+      ).to.eq(popoverElement[0].offsetHeight);
+    });
+  });
+
+  it("default picker container should not be horizontally scrollable", () => {
```

Comment:
This checks something I found: [screenshot](https://github.com/metabase/metabase/assets/6830683/cc237127-9cf6-478c-8f88-b1f45e9f34e9) (notice 2 scrollbars) and fixed in this PR.